src/backend/distributed/commands/alter_table.c

```diff
@@ -1126,10 +1128,15 @@ GetViewCreationCommandsOfTable(Oid relationId)
 		char *qualifiedViewName = quote_qualified_identifier(schemaName, viewName);
 		bool isMatView = get_rel_relkind(viewOid) == RELKIND_MATVIEW;
 
+		/* here we need to get the access method of the view to recreate it */
+		char *usingAccessMethod = CreateUsingAccessMethodClauseForMatViewIfExists(
+			viewOid);
+
 		appendStringInfo(query,
-						 "CREATE %s VIEW %s AS %s",
+						 "CREATE %s VIEW %s %s AS %s",
```

Comment:
it feels a little unexpected to append to the string even of `usingAccessMethod == NULL`. Can't we have a simple if/else?

Reply:
It's never `NULL`. Might be empty string. We can check if it's empty or not, if that makes sense?